src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+    - the ancilla is always the control qubit for the CNOT, CY, or CZ gates
+    - time step of r same as ancilla reset (always 0)
+    - time step of g same as ancilla measurement (by default 6)
+
+    ----
+    Extended RPNG format:
+
+    The extra field (a) is used together with (p) to determine the 2Q gate involving the
+    ancilla and data qubits. For example:
+    - a=z, p=x --> CNOT controlled by the ancilla and targeting the data
+    - a=z, p=z --> CZ   controlled by the ancilla and targeting the data
+    - a=y, p=z --> CY   controlled by the data    and targeting the ancilla
+    By default a=z.
+    """
+    r: Optional[ExtendedBasisEnum]
+    p: Optional[BasisEnum]
+    n: Optional[int]
+    g: Optional[ExtendedBasisEnum]
+    a: BasisEnum = BasisEnum.Z
+
+    @classmethod
+    def from_string(cls, rpng_string: str) -> 'RPNG':
+        """Initialize the RPNG object from a 4-character or 5-character string
+        
+        4-character -> simplified format -> RPNG
+        5-character -> extended format -> RAPNG
+
+        Note that the ancilla basis(a) is the second character, just before (p).
+
+        Note that any character different from a BasisEnum / ExtendedBasisEnum
+        value would result in the corresponding field being None.
+        """
+        a_str = 'z'
+        if len(rpng_string) == 4:
+            r_str, p_str, n_str, g_str = tuple(rpng_string)
+        elif len(rpng_string) == 5:
+            r_str, a_str, p_str, n_str, g_str = tuple(rpng_string)
+        else:
+            raise ValueError('The rpng string must be exactly 4- or 5-character long.')
+        # Convert the characters into the enum attributes (or raise error).
+        r = ExtendedBasisEnum(r_str) if r_str in ExtendedBasisEnum._value2member_map_ else None
+        a = BasisEnum(a_str) if a_str in BasisEnum._value2member_map_ else None
+        p = BasisEnum(p_str) if p_str in BasisEnum._value2member_map_ else None
+        n = int(n_str) if n_str.isdigit() else None
+        g = ExtendedBasisEnum(g_str) if g_str in ExtendedBasisEnum._value2member_map_ else None
+        # Raise error if anythiong but '-' was used to indicate None.
+        if not r and r_str != '-':
+            raise ValueError('Unacceptable character for the R field.')
+        if not p and p_str != '-':
+            raise ValueError('Unacceptable character for the P field.')
+        if p and not a:
+            raise ValueError('Unacceptable ancilla basis in the extended format (i.e. rapng).')
+        if not n and n_str != '-':
+            raise ValueError('Unacceptable character for the N field.')
+        if not g and g_str != '-':
+            raise ValueError('Unacceptable character for the G field.')
+        return cls(r, p, n, g, a)
+
+    
+    def get_r_op(self) -> Optional[str]:
+        """Get the reset operation or Hadamard"""
+        op = self.r
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
```

Comment:
```suggestion
        elif op.value in BasisEnum:
```
should be enough.

Reply:
The suggestion does not work but alternatives may be:
`op.value in [item.value for item in BasisEnum]`
`BasisEnum(op.value) in BasisEnum`

If there is no specific issue, I will keep the original expression.